### ISSUE TITLE
Milestones

### ISSUE THREAD

**Issue by fairwell-coding** [fairwell-coding/kddm1#1]
Open working packages
	• extended preprocessing (for all three jester subdatasets of dataset1)
		○ basic statistical analysis, i.e. skew/kurtosis, distributions (uni-multi modal etc) -> insights might lead to additional preprocessing and/or some key findings for final report
		○ outlier detection (choose one well working technique)
		○ normalization (any special tuning available for recommender systems apart from what we already have?)
		○ evaluate different data imputation techniques used for our algorithms
	• matrix factorization algorithms (select 3-4 very different approaches)
		○ scikit nmf & run with all different data imputation techniques used in preprocessing
		○ modified scikit nmf from pull request which can deal with sparse input matrices
		○ algorithm implementation based on approach showed in lectures, i.e. ALS (biconvex) only using scipy.optimize as basis using a primitive gradient descent approach (e.g. own simple SGD implementation or using RMSProp/Adam from tensorflow/pytorch)
		○ Netflix winning NMF algorithm: http://ieeexplore.ieee.org/xpl/articleDetails.jsp?arnumber=5197422
		○ Find matrix factorization algorithms in library/paper/any valid sources to compare entirely different approaches (in contrast to NMF) -> conceptual difference with pros/cons & final metrics
			§ sparsity
			§ orthogonality
			§ … maybe I still get better input what might be suitable here from the Prof
	• evaluation
		○ prepare some plots with everything needed in methods which can be reused for all algorithms & final presentation
presentation (comes later)

**Comment by elChango** [comment]
not needed anymore, see other tasks
